### PR TITLE
Fix #746

### LINKED ISSUE #746: load() not returning the config anymore broke my application
#735 stopped load from returning the configuration

### PR DESCRIPTION
**Pull request by oetiker**
load should return the current configuration

### CODE REVIEW

File: source/class/qx/tool/cli/api/AbstractApi.js [hunk 1]
```diff
@@ -32,15 +32,20 @@ qx.Class.define("qx.tool.cli.api.AbstractApi", {
     rootDir: {
       check: "String",
       nullable: false
-    }
+    },
+    /** Configuration data for the compiler */
+    configuration: {
+    },
```
Comment (oetiker) [inline source/class/qx/tool/cli/api/AbstractApi.js:38]:
```suggestion
    }
```

**Comment by hkollmann** [comment]
Here the log from the test:
Error: Property configuration of an instance of qx.tool.cli.api.AbstractApi is not (yet) ready!
    at wrapper.eval [as getConfiguration] (eval at __P_96_13 (/home/runner/work/qooxdoo-compiler/qooxdoo-compiler/lib/compiler/index.js:13154:36), <anonymous>:3:166)

getConfiguration is called before the class is constructed completely. I remember - that's was the reason for the breaking change.
```
await load();
let config = this,getConfiguration()
```
should work

File: source/class/qx/tool/cli/api/AbstractApi.js [hunk 2]
```diff
@@ -32,15 +32,20 @@ qx.Class.define("qx.tool.cli.api.AbstractApi", {
     rootDir: {
       check: "String",
       nullable: false
+    },
+    /** Configuration data for the compiler */
+    configuration: {
```
Comment (oetiker) [inline source/class/qx/tool/cli/api/AbstractApi.js:37]:
```suggestion
    configuration: {
      init: {}
```

**Comment by oetiker** [comment]
while investigating this one, I found that node_modules contains: `node_modules/@qooxdoo/compiler/node_modules/@qooxdoo/compiler/source/class/qx/tool/cli/api/AbstractApi.js` which it probably should not, AND which is even from an older version of the code ...

**Comment by oetiker** [comment]
> Here the log from the test:
> Error: Property configuration of an instance of qx.tool.cli.api.AbstractApi is not (yet) ready!
> at wrapper.eval [as getConfiguration] (eval at __P_96_13 (/home/runner/work/qooxdoo-compiler/qooxdoo-compiler/lib/compiler/index.js:13154:36), :3:166)
> 
> getConfiguration is called before the class is constructed completely. I remember - that's was the reason for the breaking change.
> 
> ```
> await load();
> let config = this.getConfiguration()
> ```
> 
> should work

well, I guess there is a better way ....

**Comment by hkollmann** [comment]
I tried it half a day but did not find one.
Good luck!

**Pull request merged**